### PR TITLE
Treat text markup (italic, bold, monofont) as blocks

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Previously, I had treated text markup (italic, bold, monofont) as some sort of special cases;  now I think they're best thought of as blocks.

### CODE REVIEW

File: doc/rdoc/markup_reference.rb [hunk 1]
```diff
@@ -194,27 +218,32 @@
...
-# - Headings.
-# - Horizontal rules.
-#
-# Note that, unlike verbatim text, single newlines are not honored,
-# but that a double newline begins a new paragraph in the block quote.
+# - {Paragraphs}[rdoc-ref:RDoc::MarkupReference@Paragraphs].
+# - {Verbatim text blocks}[rdoc-ref:RDoc::MarkupReference@Verbatim+Text+Blocks].
+# - {Code blocks}[rdoc-ref:RDoc::MarkupReference@Code+Blocks].
+# - {Lists}[rdoc-ref:RDoc::MarkupReference@Lists]:
+#   bullet, numbered, lettered, labeled.
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:242]: Do we need to specify what kinds of list can be used in block quotes? It seems like all kinds of lists can be used.

File: doc/rdoc/markup_reference.rb [hunk 2]
```diff
@@ -194,27 +218,32 @@
...
+# - {Headings}[rdoc-ref:RDoc::MarkupReference@Headings].
+# - {Horizontal rules}[rdoc-ref:RDoc::MarkupReference@Horizontal+Rules].
+# - {Text Markup}[rdoc-ref:RDoc:MarkupReference@Text+Markup]:
+#   italic, bold, monofont.
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:246]: Same with this. We can have all kinds of text markup in it, so I don't see why we need to specify that here.

File: doc/rdoc/markup_reference.rb [hunk 3]
```diff
@@ -257,7 +288,7 @@
 #   # - Another.
 #   # - An item spanning
 #   #   multiple lines.
-#   #
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:260]: Why was this line changed?

File: doc/rdoc/markup_reference.rb [hunk 4]
```diff
@@ -584,83 +615,197 @@
...
+# - Italic.
+# - Bold.
+# - Monofont.
 #
-# A single word may be italicized by prefixed and suffixed underscores.
+# Text markup may contain only one type of nested block:
 #
-# Examples:
+# - More text markup:
+#   italic, bold, monofont.
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:629]: Nit: I'm not a fan of lists with a single element. I think these should just be a single sentence.

File: doc/rdoc/markup_reference.rb (hunk 1 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb]: Done.

File: doc/rdoc/markup_reference.rb (hunk 2 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb]: Done.

File: doc/rdoc/markup_reference.rb (hunk 3 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb]: Restored.

File: doc/rdoc/markup_reference.rb (hunk 4 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb:627]: There are two of these;  I used a bullet to be parallel with multi-item lists of nested blocks.

File: doc/rdoc/markup_reference.rb [hunk 5]
```diff
@@ -98,12 +112,15 @@
...
-# - Code blocks.
-# - Block quotes.
-# - Lists of any type.
-# - Headings.
-# - Horizontal rules.
+# - {Verbatim text blocks}[rdoc-ref:RDoc::MarkupReference@Verbatim+Text+Blocks].
+# - {Code blocks}[rdoc-ref:RDoc::MarkupReference@Code+Blocks].
+# - {Block quotes}[rdoc-ref:RDoc::MarkupReference@Block+Quotes].
+# - {Lists}[rdoc-ref:RDoc::MarkupReference@Lists]:
+#   bullet, numbered, lettered, labeled.
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:119]: Ditto here, I think we can removed the kinds of lists.

File: doc/rdoc/markup_reference.rb [hunk 6]
```diff
@@ -240,12 +267,14 @@
...
-# - Headings.
-# - Horizontal rules.
+# - {Paragraphs}[rdoc-ref:RDoc::MarkupReference@Paragraphs].
+# - {Verbatim text blocks}[rdoc-ref:RDoc::MarkupReference@Verbatim+Text+Blocks].
+# - {Code blocks}[rdoc-ref:RDoc::MarkupReference@Code+Blocks].
+# - {Block quotes}[rdoc-ref:RDoc::MarkupReference@Block+Quotes].
+# - {Headings}[rdoc-ref:RDoc::MarkupReference@Headings].
+# - {Horizontal rules}[rdoc-ref:RDoc::MarkupReference@Horizontal+Rules].
+# - {Text Markup}[rdoc-ref:RDoc:MarkupReference@Text+Markup]:
+#   italic, bold, monofont.
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:277]: Ditto here, we can remove the types of text markup.

File: doc/rdoc/markup_reference.rb [hunk 7]
```diff
@@ -370,30 +399,30 @@
 #   # ============Still a Heading (Level 6)
 #   # \== Not a Heading
 #
+# A heading may contain only one type of nested block:
+#
+# - {Text Markup}[rdoc-ref:RDoc:MarkupReference@Text+Markup]:
+#   italic, bold, monofont.
```
Comment (peterzhu2118) [inline doc/rdoc/markup_reference.rb:405]: Also ditto here, we can remove the types of text markup.

File: doc/rdoc/markup_reference.rb (hunk 5 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb]: Done.

File: doc/rdoc/markup_reference.rb (hunk 6 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb]: Done.

File: doc/rdoc/markup_reference.rb (hunk 7 above)
Comment (BurdetteLamar) [reply doc/rdoc/markup_reference.rb]: Done.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Comment by BurdetteLamar** [comment]
How did this get closed?  Is there a problem?  Anyone?
